src/symbolic_value.ml

```diff
@@ -113,63 +78,34 @@ module Ref = struct
 end
 
 module Bool = struct
-  let of_val = function
-    | Val True -> Some true
-    | Val False -> Some false
-    | _ -> None
+  let const b = Bool.v b
+
+  let not e = Bool.not_ e
 
-  let const b = Val (if b then True else False) @: Ty_bool
-
-  let not e =
-    match e.node.e with
-    | Unop (Not, cond) -> cond
-    | e' ->
-      Option.value ~default:(Unop (Not, e) @: Ty_bool)
-      @@ let+ b = of_val e' in
-         const (not b)
-
-  let or_ e1 e2 =
-    match (of_val e1.node.e, of_val e2.node.e) with
-    | Some b1, Some b2 -> const (b1 || b2)
-    | Some false, _ -> e2
-    | _, Some false -> e1
-    | Some true, _ | _, Some true -> const true
-    | _ -> Binop (Or, e1, e2) @: Ty_bool
-
-  let and_ e1 e2 =
-    match (of_val e1.node.e, of_val e2.node.e) with
-    | Some b1, Some b2 -> const (b1 && b2)
-    | Some true, _ -> e2
-    | _, Some true -> e1
-    | Some false, _ | _, Some false -> const false
-    | _ -> Binop (And, e1, e2) @: Ty_bool
+  let or_ e1 e2 = Bool.or_ e1 e2
+
+  let and_ e1 e2 = Bool.and_ e1 e2
```

Comment:
We could `include Bool` to avoid redefining all these ? I also prefer using `v` to `const` so I'm fine with changing it in Owi

Reply:
Certainly, I can do that. However, I'm concerned about simply changing `Value.Bool.const` to `Value.Bool.v` because it would no longer align with the consistency of value constructors for constants, such as `Value.const_i32`, `Value.const_i64`, `Value.const_f32`, and `Value.const_f64`.

One solution is to move these constructors into their respective modules, transforming, for example, `Value.const_i32` into `Value.I32.v`. 

What would you prefer?